kernel/pci/src/lib.rs

```diff
@@ -186,7 +186,7 @@ fn scan_pci() -> Vec<PciBus> {
 
 /// The bus, slot, and function number of a given PCI device.
 /// This offers methods for reading and writing the PCI config space. 
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, PartialEq)]
```

Comment:
good call, while you're at it you can derive `Eq` too, and `Hash` if we want. Those typically go together.